### PR TITLE
Bump minimum Kokkos version from 3.4 to 3.5

### PR DESCRIPTION
**Pull request by dalg24**
Kokkos release 3.6 is out.  We had decided we would only support the last two releases.
Potential downside is ArborX may not build against the latest Trilinos release.
This change allow us to enable the minimum spanning tree unconditionally (required Desul atomics) and to drop our detection idiom facility.

### CODE REVIEW

**Review by aprokop — APPROVED** [review]
LGTM. Needs a matching PR to the Wiki to update the testing page.

**Comment by dalg24** [comment]
Need to deal with 2 issues before we can merge:
* issue with NVIDIA images and [gpg repo key rotation](https://developer.nvidia.com/blog/updating-the-cuda-linux-gpg-repository-key/)
* issue with half precision type and older CUDA version

File: .jenkins/continuous.groovy [hunk 1]
```diff
@@ -36,131 +36,12 @@ pipeline {
...
-                        }
-                    }
-                }
                 stage('CUDA-10.0-Clang') {
                     agent {
                         dockerfile {
                             filename "Dockerfile"
                             dir "docker"
-                            additionalBuildArgs '--build-arg BASE=nvidia/cuda:10.0-devel-ubuntu18.04 --build-arg KOKKOS_OPTIONS="-DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_COMPILER=clang++ -DKokkos_ENABLE_PTHREAD=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON -DKokkos_ARCH_VOLTA70=ON"'
+                            additionalBuildArgs '--build-arg BASE=nvidia/cuda:10.0-devel-ubuntu18.04 --build-arg KOKKOS_OPTIONS="-DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_COMPILER=clang++ -DKokkos_ENABLE_PTHREAD=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON -DKokkos_ARCH_VOLTA70=ON" --build-arg KOKKOS_VERSION=develop'
```
Comment (masterleinad) [inline .jenkins/continuous.groovy:44]:
```suggestion
                            additionalBuildArgs '--build-arg BASE=nvidia/cuda:10.0-devel-ubuntu18.04 --build-arg KOKKOS_OPTIONS="-DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_COMPILER=clang++ -DKokkos_ENABLE_THREADS=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ENABLE_CUDA_LAMBDA=ON -DKokkos_ARCH_VOLTA70=ON" --build-arg KOKKOS_VERSION=develop'
```
?

**Comment by aprokop** [comment]
Needs to be rebase on `master`.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by aprokop** [comment]
One of those "CI passed, but was marked as failed" thing. Merging.

File: .jenkins/continuous.groovy (hunk 1 above)
Comment (dalg24) [reply .jenkins/continuous.groovy]: Uggg that was a mistake.  That builds uses Kokkos 3.5
This disabled the threads backend effectively...

**Pull request merged**